### ISSUE TITLE
0.5.0.b0: AttributeError: module 'charms.reactive.bus' has no attribute 'StateList'

### ISSUE THREAD

**Issue by ajkavanagh** [juju-solutions/charms.reactive#125]
The manila charm uses (a possibly little used) feature which is the `StateList`.  Error from install hook:

```
2017-09-18 15:23:58 DEBUG install     class ManilaPluginProvides(reactive.RelationBase):
2017-09-18 15:23:58 DEBUG install   File "/var/lib/juju/agents/unit-manila-0/charm/hooks/relations/manila-plugin/provides.py", line 39, in ManilaPluginProvides
2017-09-18 15:23:58 DEBUG install     class states(reactive.bus.StateList):
2017-09-18 15:23:58 DEBUG install AttributeError: module 'charms.reactive.bus' has no attribute 'StateList'
2017-09-18 15:23:58 ERROR juju.worker.uniter.operation runhook.go:107 hook "install" failed: exit status 1
```

The manila charm is probably the only openstack reactive charm that uses the StateList.  Should manila be refactored to remove it, or should the support be put back into `charms.reactive`?

**Comment by johnsca** [comment]
The `StateList` class is [still available](https://github.com/juju-solutions/charms.reactive/blob/master/charms/reactive/flags.py#L21), it was just moved and deprecated because I didn't think it really add anything useful and wasn't being used.  If that's not true, we can discuss how to improve it and undeprecate it.

I didn't import a reference back in to `charms.reactive.bus` because the new `charms.reactive.flags` has a dependency on `charms.reactive.bus` which would create an import loop.  I definitely feel like it makes more sense in the `flags` package, so if it's only one charm using it, I'd prefer to refactor that to either use the new location or not use it at all.

**Comment by ajkavanagh** [comment]
@johnsca I can only confirm that only one "OpenStack" charm is using the feature.  I have no idea if other, non-OpenStack, charms written by other groups/developers/ops might be using it or not.  I think it was/used to be referenced in the documentation as a way to declare the states the Interface used.

**Comment by johnsca** [comment]
AFAICT it's only mentioned in its own API docs.  I like the idea of the of having the flags be more consistently documented, but it seems a little late to start a convention that requires active participation at this point.

**Comment by ajkavanagh** [comment]
I don't disagree.  I've raised a bug on charm-manila (actually it's in the interface manila-plugin) to get it fixed when 0.5.0 is released. https://bugs.launchpad.net/charm-manila/+bug/1718152
